sklearn/feature_selection/tests/test_mutual_info.py

```diff
@@ -168,12 +176,18 @@ def test_mutual_info_classif_mixed():
         assert mi_nn[2] == mi[2]
 
 
-def test_mutual_info_options():
-    X = np.array([[0, 0, 0], [1, 1, 0], [2, 0, 1], [2, 0, 1], [2, 0, 1]], dtype=float)
-    y = np.array([0, 1, 2, 2, 1], dtype=float)
+def test_mutual_info_options(global_dtype):
+    X = np.array(
+        [[0, 0, 0], [1, 1, 0], [2, 0, 1], [2, 0, 1], [2, 0, 1]], dtype=global_dtype
+    )
+    y = np.array([0, 1, 2, 2, 1])
     X_csr = csr_matrix(X)
 
-    for mutual_info in (mutual_info_regression, mutual_info_classif):
+    for mutual_info, y_dtype in (
+        (mutual_info_regression, global_dtype),
+        (mutual_info_classif, y.dtype),
+    ):
+        y = y.astype(global_dtype)
```

Comment:
why don't you set the dtype right away ?

Reply:
So as to preserve int for classification.